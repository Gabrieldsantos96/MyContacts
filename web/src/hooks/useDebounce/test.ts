import { renderHook } from '@testing-library/react'
import { useDebounce } from '.'

vitest.useFakeTimers()

describe('useDebounce', () => {
  test('returns the initial value', () => {
    const { result } = renderHook(() => useDebounce({ value: 'initialValue' }))
    expect(result.current.debouncedValue).toBe('initialValue')
  })
})
