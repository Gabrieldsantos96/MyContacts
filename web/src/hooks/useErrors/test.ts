import { renderHook, act } from '@testing-library/react'
import useErrors from '.'

describe('useErrors', () => {
  test('initializes with no errors', () => {
    const { result } = renderHook(() => useErrors())

    expect(result.current.errors).toHaveLength(0)
    expect(result.current.noErrors).toBe(true)
  })

  test('adds and removes errors', () => {
    const { result } = renderHook(() => useErrors())

    const errorToAdd = { field: 'username', message: 'Username is required' }

    act(() => {
      result.current.setError([errorToAdd], 'username')
    })

    expect(result.current.errors).toEqual([errorToAdd])
    expect(result.current.noErrors).toBe(false)

    act(() => {
      result.current.removeError('username')
    })

    expect(result.current.errors).toHaveLength(0)
    expect(result.current.noErrors).toBe(true)
  })

  test('replaces existing errors for a field', () => {
    const { result } = renderHook(() => useErrors())

    const initialError = { field: 'email', message: 'Email is required' }
    const updatedError = { field: 'email', message: 'Invalid email format' }

    act(() => {
      result.current.setError([initialError], 'email')
    })

    expect(result.current.errors).toEqual([initialError])

    act(() => {
      result.current.setError([updatedError], 'email')
    })

    expect(result.current.errors).toEqual([updatedError])
  })

  test('gets error message by field name', () => {
    const { result } = renderHook(() => useErrors())

    const error = {
      field: 'password',
      message: 'Password must be at least 8 characters'
    }

    act(() => {
      result.current.setError([error], 'password')
    })

    const errorMessage = result.current.getErrorMessageByFieldName('password')

    expect(errorMessage).toBe(error.message)
  })
})
