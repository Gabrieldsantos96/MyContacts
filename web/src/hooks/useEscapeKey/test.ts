import { renderHook, act } from '@testing-library/react'
import { useEscapeKey } from '.'
import React from 'react'

describe('useModal', () => {
  it('should call when press escape', () => {
    const onCancelMock = vitest.fn()
    renderHook(() => useEscapeKey(onCancelMock))

    act(() => {
      const escapeEvent = new KeyboardEvent('keydown', { key: 'Escape' })
      document.body.dispatchEvent(escapeEvent)
    })

    expect(onCancelMock).toHaveBeenCalled()
  })

  it('should not call for other keys', () => {
    const onCancelMock = vitest.fn()

    renderHook(() => useEscapeKey(onCancelMock))

    act(() => {
      const otherKeyEvent = new KeyboardEvent('keydown', { key: 'Enter' })
      document.body.dispatchEvent(otherKeyEvent)
    })

    expect(onCancelMock).not.toHaveBeenCalled()
  })

  it('Should run useEffect cleanUp return function', () => {
    let isExecutedAfterUnmount = false

    vitest.spyOn(React, 'useEffect').mockImplementationOnce(() => {
      isExecutedAfterUnmount = true
    })

    const { unmount } = renderHook(() => useEscapeKey(vitest.fn()))

    unmount()

    expect(isExecutedAfterUnmount).toBe(true)
  })
})
