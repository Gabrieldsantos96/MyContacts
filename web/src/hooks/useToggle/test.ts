import { renderHook, act } from '@testing-library/react'
import { useToggle } from '.'

describe('useToggle', () => {
  test('initializes with default value', () => {
    const { result } = renderHook(() => useToggle(true))

    expect(result.current.value).toBe(true)
  })

  test('initializes with default value as false when not provided', () => {
    const { result } = renderHook(() => useToggle())

    expect(result.current.value).toBe(false)
  })

  test('toggles the value', () => {
    const { result } = renderHook(() => useToggle(true))

    act(() => {
      result.current.toggle()
    })

    expect(result.current.value).toBe(false)
  })

  test('sets the value directly', () => {
    const { result } = renderHook(() => useToggle(true))

    act(() => {
      result.current.setValue(false)
    })

    expect(result.current.value).toBe(false)
  })
})
