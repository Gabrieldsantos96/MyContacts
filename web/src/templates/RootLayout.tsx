import { Outlet } from 'react-router-dom'
import Header from '../components/Header'
import * as S from './RootLayoutStyles'

const RootLayout = () => (
  <S.Container>
    <Header />
    <Outlet />
  </S.Container>
)

export { RootLayout }
