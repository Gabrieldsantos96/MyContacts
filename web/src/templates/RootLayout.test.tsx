import { render, screen } from '@utils/test/test-ui'

import { RootLayout } from './RootLayout'

vitest.mock('react-router-dom', () => ({
  __esModule: true,
  Outlet: function Mock() {
    return <div data-testid="outlet" />
  }
}))

vitest.mock('@components/Header', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="header" />
  }
}))

describe('<RootLayout/>', () => {
  it('should render RootLayout', async () => {
    const { container } = render(<RootLayout />)

    expect(screen.getByTestId('header')).toBeInTheDocument()

    expect(screen.getByTestId('outlet')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
