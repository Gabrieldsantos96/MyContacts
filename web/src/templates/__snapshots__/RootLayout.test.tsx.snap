// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RootLayout/> > should render RootLayout 1`] = `
<div
  class="sc-beySPh DCYkH"
>
  <div
    data-testid="header"
  />
  <div
    data-testid="outlet"
  />
</div>
`;
